wp_api/src/login/login_client.rs

```diff
@@ -329,19 +329,49 @@ impl WpLoginClient {
     ) -> Result<WpNetworkResponse, ParseHtmlFailure> {
         let site_url = ParsedUrl::parse(attempt_site_url)
             .map_err(|error| ParseHtmlFailure::ParseSiteUrl { error })?;
-        self.request_executor
-            .execute(
-                WpNetworkRequest {
-                    uuid: Uuid::new_v4().into(),
-                    retry_count: 0,
-                    method: RequestMethod::GET,
-                    url: WpEndpointUrl(site_url.url()),
-                    header_map: WpNetworkHeaderMap::default().into(),
-                    body: None,
+        self.perform(
+            WpNetworkRequest {
+                uuid: Uuid::new_v4().into(),
+                retry_count: 0,
+                method: RequestMethod::GET,
+                url: WpEndpointUrl(site_url.url()),
+                header_map: WpNetworkHeaderMap::default().into(),
+                body: None,
+            }
+            .into(),
+        )
+        .await
+        .map_err(|error| ParseHtmlFailure::FetchSite { error })
+    }
+
+    async fn perform(
+        &self,
+        request: Arc<WpNetworkRequest>,
+    ) -> Result<WpNetworkResponse, RequestExecutionError> {
+        let mut retry_count = 0;
+
+        loop {
+            let response = self.request_executor.execute(request.clone()).await?;
+
+            if retry_count >= 3 {
+                return Err(RequestExecutionError::RequestExecutionFailed {
+                    status_code: Some(response.status_code),
+                    redirects: None,
+                    reason: RequestExecutionErrorReason::MisconfiguredRateLimitError {},
+                });
+            }
+
+            if response.is_http_429() {
+                let retry_after = response.get_retry_after();
```

Comment:
Make sure it's the value is in a given range? So that, we don't retry too soon and too often, also don't wait forever.